src/client/opamClient.ml

```diff
@@ -315,6 +315,63 @@ let upgrade_t ?strict_upgrade ?auto_install ?ask ?(check=false) ~all atoms t =
              (OpamPackage.names_of_packages t.compiler_packages)
          in
          let unopt = unopt -- base in
+         let get_formula pkg get =
+           OpamStd.Option.map
+             (fun opam ->
+                OpamFilter.filter_formula ~default:false
+                  (OpamFilter.deps_var_env ~build:true ~post:false
+                     ~test:true ~dev:false ~doc:false)
+                  (get opam))
+             (OpamSwitchState.opam_opt t pkg)
+         in
+         let conflicts =
+           OpamPackage.Set.fold (fun unopt_pkg map ->
+               let set =
+                 OpamSwitchState.conflicts_with t
+                   (OpamPackage.Set.singleton unopt_pkg) latest in
+               OpamPackage.Set.fold (fun installed_pkg map1 ->
+                   match get_formula installed_pkg  OpamFile.OPAM.conflicts with
+                   | None -> map1
+                   | Some conflicts_formula ->
+                     if OpamFormula.verifies conflicts_formula unopt_pkg then
+                       OpamFormula.fold_left
+                         (fun map2 (n,formula) ->
```

Comment:
Why use a different name (`map2`) if it's always the same accumulator being passed among the different depth levels ?

Reply:
Personal preference :)

Reply:
I'd argue that you risk mixing the different `acc`s and using the wrong one, since they all have the same type, which would cause hard-to-spot bugs, while you are safer with using the same name along the chain with masking ;)